lib/queue.rb

```diff
@@ -1,31 +1,46 @@
 class Queue
-
-  def initialize
-    # @store = ...
-    raise NotImplementedError, "Not yet implemented"
+  def initialize(size = 10)
+    @store = Array.new(size)
+    @front = @back = 0
+    @empty = true
   end
 
   def enqueue(element)
-    raise NotImplementedError, "Not yet implemented"
+    if @front == @back && !@empty
+      # this is quen the queue is full, throw error 
+      raise ArgumentError
+    else
+      @store[@back] = element
+      @empty = false
+      @back = (@back + 1) % @store.length
+    end
   end
 
   def dequeue
-    raise NotImplementedError, "Not yet implemented"
+    value = @store[@front]
+    @store[@front] = nil
+    @front = (@front + 1) % @store.length
+    if @front == @back
+      @empty = true
+    end
+    return value
   end
 
   def front
-    raise NotImplementedError, "Not yet implemented"
+    return @store[@front]
   end
 
   def size
-    raise NotImplementedError, "Not yet implemented"
+    return @store.size
   end
 
   def empty?
-    raise NotImplementedError, "Not yet implemented"
+    return @empty
   end
 
   def to_s
     return @store.to_s
   end
```

Comment:
This will leave multiple `nil` elements in the output.  Something like this would work.

```suggestion
  def to_s
    return [@front... @back]  if @front < @back
    return [0...@back] + [@front...@store.length]
  end
```